csunplugged/resources/views/views.py

```diff
@@ -150,6 +153,12 @@ def generate_resource_copy(generator):
     elif paper_size == "letter":
         max_pixel_height = 249 * MM_TO_PIXEL_RATIO
 
+    if thumbnail and len(data) > 1:
+        try:
+            data = [next(page for page in data if page.get("thumbnail", False))]
```

Comment:
This is pretty hard to parse. Perhaps consider

`data = list(filter(lambda data: data.get("thumbnail"), data))`

This will filter the list to only include pages where thumbnail is True. You can then do a length check, and raise an error if empty (no thumbnail listed) or more than one page was marked.